File: 백준/Bronze/1032. 명령 프롬프트/명령 프롬프트.rb
n = gets.to_i
pattern = gets.chomp

(n-1).times do 
	file = gets.chomp
	file.each_char.with_index do |ch, i|
		if pattern[i] != file[i]
			pattern[i] = "?"
		end
	end
end 
puts pattern

File: 백준/Bronze/1032. 명령 프롬프트/명령 프롬프트.cc
#include <iostream>
using namespace std;

int main() {
	int n;
	cin >> n;
	string pattern;
	cin >> pattern;
	
	for(int i=0; i<n-1; ++i) {
		string file;
		cin >> file;
		
		for(int j=0; j<file.size(); ++j) {
			if(pattern[j] != file[j]) {
				pattern[j] = '?';
			}
		}
	}
	
	cout << pattern << endl;
	
	return 0;
}